woocommerce/woocommerce-gutenberg-products-block
# Add an endpoint for updating customer by order

Fixes #9461 

#### Other Checks


### Screenshots

<img width="512" alt="Screen Shot 2023-07-09 at 6 42 20 PM" src="https://github.com/woocommerce/woocommerce-blocks/assets/56378160/612e8c04-99f1-44d4-abd2-d357f5c13f15">


#### Automated Tests

#### Testing instruction
* Test checkout as a guest*
1. Install and activate a plugin with pay-for-order flow. ex WooCommerce-Bookings
2. Create a product that needs confirmation
3. Purchase the product with pay-for-order flow
1. Go to WooCommerce -> Orders
1. Make sure the Customer is `Guest` 
1. Click the `Customer payment page` to go to the pay-for-order page
1. Copy the `order_id` and the `key` from the URL
1. Use Postman to test this endpoint http://merchant.local/wp-json/wc/store/order/YOUR_ORDER_NUMBER/update-customer?key=YOUR_KEY&billing_email=YOUR_BILLING_EMAIL
2. Make sure on Postman the authorization type set to `No Auth` 
3. POST to update billing & shipping address. Ex. last name
```
{
  "billing_address": {
    "last_name": "Guest"
  },
  "shipping_address": {
    "last_name": "Guest"
  }
}
```
11. Check that the billing address and the shipping address are updated

*Test checkout as an existing customer*
1. Use the same order you created above
9. Change the Customer to an existing customer
3. Go to WooCommerce -> Settings -> Advanced -> Rest API
4. Add key with the customer email, read/write permission, copy the Consumer key and secret
5. Go to the same request on Postman, change the authorization type to `OAuth 1.0`, and paste the key and secret there
6. Remove the `key` and `billing_email` params
7. Update the POST body to a new last name so you can see the update later 
8. Send the request
9. Go back to WooCommerce -> Orders -> your order
10. Check the billing and shipping last names are updated



### Changelog

> Add an endpoint for updating customers by order
